plugins/modules/packaging/language/maven_artifact.py

```diff
@@ -509,7 +516,13 @@ def _request(self, url, failmsg, force=True):
         self.module.params['url_password'] = self.module.params.get('password', '')
         self.module.params['http_agent'] = self.user_agent
 
-        response, info = fetch_url(self.module, url_to_use, timeout=req_timeout, headers=self.headers)
+        response, info = fetch_url(
+            self.module,
+            url_to_use,
+            timeout=req_timeout,
+            headers=self.headers,
+            unredirected_headers=self.module.params.get('unredirected_headers', ['Authorization', 'Cookies'])
```

Comment:
```suggestion
            unredirected_headers=self.module.params['unredirected_headers'],
```
Also `undirected_headers` must only be passed to `fetch_url` if it has content. So better use
```suggestion
            **kwargs
```
and initialize `kwargs = {}` first, and only put in `undirected_headers` if the option was specified / has non-trivial content.